DataProcessing/datax-host/src/main/scala/datax/processor/CommonProcessorFactory.scala

```diff
@@ -608,8 +618,18 @@ object CommonProcessorFactory {
           val blobStorageKey = ExecutorHelper.createBlobStorageKeyBroadcastVariable(paths.head, spark)
 
           val inputDf = internalFiles
-            .flatMap(file => HadoopClient.readHdfsFile(file.inputPath, gzip = file.inputPath.endsWith(".gz"), blobStorageKey)
-              .filter(l => l != null && !l.isEmpty).map((file, outputPartitionTime, _)))
+            .flatMap(file => {
+              val timeLast = System.nanoTime()
+              val retVal = HadoopClient.readHdfsFile(file.inputPath, gzip = file.inputPath.endsWith(".gz"), blobStorageKey)
+                .filter(l => l != null && !l.isEmpty).map((file, outputPartitionTime, _))
+              val timeNow = System.nanoTime()
+              AppInsightLogger.trackEvent(
+                ProductConstant.ProductRoot + "/ReadBlob",
+                Map("Timestamp" -> timeNow.toString, "InputPath" -> file.inputPath),
+                Map("ReadTime" -> (timeNow - timeLast) / 1E9)
```

Comment:
in addition to read time, can we log # of items loaded?

Reply:
events logged if easy to get

Reply:
This event is scoped to individual files, we got another metric that outputs the number of input blobs